partiql-parser/src/partiql.pest

```diff
@@ -14,9 +14,39 @@ Scanner = _{ SOI ~ Token }
 Token = _{
       Keyword
     | String
+    | Number
     | Identifier
 }
 
+//
+// Numeric Literals
+//
+
+Number = ${
+      DecimalExp
+    | Decimal
+    | Integer
+}
```

Comment:
The `$` means *compound atomic*.  Which means that sub-rules can produce pairs, but whitespace does not apply, which we definitely don't want in number parsing.

See: https://docs.rs/pest_derive/2.1.0/pest_derive/#modifiers